### PR TITLE
chore(internal): add HTTP connector utility

### PR DESCRIPTION
**Pull request by P403n1x87**
This change adds an HTTP connector utility that wraps a Python
connection object with a context manager for convenience and reuse.
The context manager essentially closes over the connection information
so that new connections to the same socket can be created by simply
calling the result of the connector helper. Examples are provided by the
new test cases.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by majorgreys — DISMISSED** [review]

**Review by mabdinur — DISMISSED** [review]

File: tests/internal/test_utils_http.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from ddtrace.internal.utils.http import connector
+from tests.utils import MockHTTPServer
+
+
+def test_connector():
+    with MockHTTPServer(port=8181) as server:
```
Comment (brettlangdon) [inline tests/internal/test_utils_http.py:6]: do you want a real server running or would `HTTPretty` be good enough?

File: tests/internal/test_utils_http.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            def do_GET(self):
+                self.respond(
+                    200,
+                    headers={"Content-Type": "application/json"},
+                    body='{"hello": "world"}',
+                )
+
+        server.add_route("/api/test", TestHandler)
+
+        connect = connector("http://localhost:8181")
```
Comment (brettlangdon) [inline tests/internal/test_utils_http.py:18]: we should have tests for unix domain sockets as well.

File: tests/internal/test_utils_http.py (hunk 1 above)
Comment (P403n1x87) [reply tests/internal/test_utils_http.py:6]: In theory `HTTPretty` might be enough for this.

**Review by majorgreys — APPROVED** [review]
Even better!

**Review by mabdinur — APPROVED** [review]

**Pull request merged**